### ISSUE TITLE
Meta issue: Terraform files other than .tf

### ISSUE THREAD

**Issue by danieldreier** [hashicorp/vscode-terraform#636]
The purpose of this issue is to track requests for editor support of non-TF file types, including tfvars files, templates, .terraformrc, etc. This is a thematic issue to support planning, and will be broken down into more concrete tasks before starting work.

**Comment by gruckion** [comment]
Any updates... seems like quite an easy win and import to have. Looking at my `*.tftpl` JSON file in black and white isn't very nice.

**Comment by freeman** [comment]
It seems the extension is not supporting `.tf.json` files as well

**Comment by radeksimko** [comment]
I think there may be differences in opinions on what "support" really means.

As you pointed out, the language server does parse JSON files to provide to provide e.g. completion of references to configuration inside those JSON files. It does not however provide any code intelligence (completion, hover etc.) _inside_ those JSON files.

As it turns out this level of support is not ideal - @dbanck points out some downsides in https://github.com/hashicorp/terraform-ls/issues/705 - so it's clear we'll have to make some changes in this area. That said I am still not sure that providing completion/hover etc. _inside_ JSON files is what we really should be offering as the JSON format is primarily for machines to read, not humans, so I wouldn't expect humans to be hand-editing JSON files.

TL;DR @freeman Can you clarify what "not supporting" means to you?

**Comment by freeman** [comment]
@radeksimko In my project I have `.tf.json` files containing data sources (not `.tfvars.json`) and I get no completion from the language server in VS Code.

**Comment by radeksimko** [comment]
@freeman That sounds like a bug - we should already support such a scenario, would you mind [creating a separate bug report](https://github.com/hashicorp/vscode-terraform/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml) and attaching some more details, ideally with pieces of configuration, so we can reproduce and look into it, and hopefully get it fixed? 🙏🏻

**Comment by flcdrg** [comment]
The current docs imply that using `.tfbackend` will "likely provide better editing experience". It would be great to be able to update the docs to remove the word 'likely' 😄 

https://developer.hashicorp.com/terraform/language/settings/backends/configuration#file
